webpack/index.d.ts

```diff
@@ -654,7 +654,20 @@ declare namespace webpack {
      */
 
     class BannerPlugin extends Plugin {
-        constructor(banner: any, options: any);
+        constructor(options?: BannerPlugin.Options | string);
+    }
+
+    namespace BannerPlugin {
+        type Filter = string | RegExp;
+        
+        interface Options {
+            banner?: string;
```

Comment:
`banner?` shouldn't be optional, see [here](https://github.com/webpack/webpack/blob/8a95bcc2f45c2bfa26ae06e5fbf7a81f6dd37e69/lib/BannerPlugin.js#L25)